angular-app/src/app/pages/appointments/appointment-details/appointment-details.component.ts

```diff
@@ -103,4 +103,52 @@ export class AppointmentDetailsComponent {
     });
   }  
 
+  // Funciton to book the currently signed in user
+  bookUser(appt: IAppointment) {
+    if(appt.participants.length < appt.totalSlots) {
+      this.appointmentService.bookOtherUser(appt._id as string, this.userEmail).subscribe(
+        data => { 
+          console.log(data);
+          if(data.status.toString() == '200') {
+            this.openSnackBar("Appointment successfully booked!", "Dismiss", {
+              duration: 5000,
+            });
+            appt.participants[1] = this.userEmail as string; 
+          } else {
+            this.openSnackBar("An error occured when trying to book this appointment.", "Dismiss", {
+              duration: 5000,
+            }); 
+          }
+        }
+      );
+    } else {
+      console.log("Cannot book this appointment. Something went wrong.");
+    }
+  }
+
+  unbookUser(appt: IAppointment) {
+    if(!(appt.participants[1] === undefined)) {
+      this.appointmentService.unbookOtherUser(appt._id as string, appt.participants[1] as string).subscribe(
+        data => { 
+          console.log(data);
+          if(data.status.toString() == '200') {
+            this.openSnackBar("Appointment successfully unbooked!", "Dismiss", {
+              duration: 5000,
+            }); 
+            appt.participants.pop();
```

Comment:
Good point, I'll take your suggestion. I'm banking on the fact that the `participant[0]` value will always be the talent and the `participant[1]` value will always be the participant but I added a safeguard in case that wasn't the case.